cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -167,11 +167,7 @@ protected void run() throws Exception {
         throw e;
       }
     } finally {
-      try {
-        Closeables.closeQuietly(executionContext);
-      } finally {
-        Thread.currentThread().setContextClassLoader(oldClassLoader);
-      }
```

Comment:
hum, so need to move the close to shutdown, but both contexts need to be closed.